app/assets/javascripts/discourse/app/components/topic-map/topic-map-summary.gjs

```diff
@@ -1,162 +1,421 @@
-    };
+  @action
+  showAllLinks() {
+    this.allLinksShown = true;
   }
 
   get shouldShowParticipants() {
     return (
-      this.args.collapsed &&
-      this.args.topic.posts_count > 2 &&
-      this.args.topicDetails.participants &&
-      this.args.topicDetails.participants.length > 0
+      this.args.topic.posts_count >= 10 &&
+      this.args.topicDetails.participants?.length >= 2 &&
+      !this.site.mobileView
     );
   }
 
-  get createdByAvatar() {
-    return htmlSafe(
-      avatarImg({
-        avatarTemplate: this.args.topicDetails.created_by?.avatar_template,
-        size: "tiny",
-        title:
-          this.args.topicDetails.created_by?.name ||
-          this.args.topicDetails.created_by?.username,
-      })
+  get readTime() {
+    const calculatedTime = Math.ceil(
+      Math.max(
+        this.args.topic.word_count / this.siteSettings.read_time_word_count,
+        (this.args.topic.posts_count * MIN_POST_READ_TIME) / 60
+      )
     );
+
+    return calculatedTime > 3 ? calculatedTime : null;
   }
 
-  get lastPostAvatar() {
-    return htmlSafe(
-      avatarImg({
-        avatarTemplate: this.args.topicDetails.last_poster?.avatar_template,
-        size: "tiny",
-        title:
-          this.args.topicDetails.last_poster?.name ||
-          this.args.topicDetails.last_poster?.username,
-      })
+  get topRepliesSummaryEnabled() {
+    return this.args.postStream.summary;
+  }
+
+  get topRepliesTitle() {
+    if (this.topRepliesSummaryEnabled) {
+      return;
+    }
+
+    return I18n.t("summary.short_title");
+  }
+
+  get topRepliesLabel() {
+    const label = this.topRepliesSummaryEnabled ? "All Replies" : "Top Replies";
+
+    return label;
+  }
+
+  get topRepliesIcon() {
+    if (this.topRepliesSummaryEnabled) {
+      return;
+    }
+
+    return "layer-group";
+  }
+
+  get loneStat() {
+    const hasViews = this.args.topic.views >= 0;
+    const hasLikes =
+      this.args.topic.like_count > 5 && this.args.topic.posts_count > 10;
+    const hasLinks = this.linksCount > 0;
+    const hasUsers = this.args.topic.participant_count > 5;
+    const canSummarize = this.args.topic.has_summary;
+
+    if (canSummarize) {
+      return false;
+    }
+
+    return (
+      [hasViews, hasLikes, hasLinks, hasUsers].filter(Boolean).length === 1
     );
   }
 
+  @action
+  fetchMostLiked() {
+    const cacheKey = `mostLikedPosts_${this.args.topic.id}`;
+    const cachedData = this.mapCache.get(cacheKey);
+    this.loading = true;
+
+    if (cachedData) {
+      this.mostLikedPosts = cachedData;
+      this.loading = false;
+      return;
+    }
+
+    const filter = `/search.json?q=" " topic%3A${this.args.topic.id} order%3Alikes`;
+
+    ajax(filter)
+      .then((data) => {
+        data.posts.sort((a, b) => b.like_count - a.like_count);
+        const mostLikedPosts = data.posts
+          .filter((post) => post.post_number !== 1 && post.like_count !== 0)
+          .slice(0, 3);
```

Comment:
How about doing the filtering and then the sorting?
Plus using a better variable name?

```suggestion
        const top3LikedPosts = data.posts
          .filter((post) => post.post_number > 1 && post.like_count > 0)
          .sort((a, b) => b.like_count - a.like_count)
          .slice(0, 3);
```